### DISCUSSION TITLE
ScribeOCR and its relationship to tesseract's engine

### DISCUSSION THREAD

**Discussion by ajlowy66** [scribeocr/scribeocr#45] [General]
Very elegant and well written repo/app! I just wanted to ascertain that ScribeOCR's image or PDF enhancement relies on the user's _previously_ OCR-enhanced documents (by tesseract or ABBYY Fine Reader), which they upload to ScribeOCR, which _then_ does its own enhancement/OCR—but **_without_** using or interacting with the user's existing tesseract files in their `tesseract-ocr` directory, especially the language-specific `traineddata` files in its `tessdata` subdirectory.

As I found my way to ScribeOCR by virtue of its appearance in the list—provided by [Tesseract's User Manual on GitHub ("GUIs and Other Projects using Tesseract OCR")](https://tesseract-ocr.github.io/tessdoc/User-Projects-%E2%80%93-3rdParty.html)—of front-end apps using tesseract as their OCR engine, I assumed that ScribeOCR operated that way as well.

However, I only realised it was probably independent of tesseract when I discovered ScribeOCR only offers a tiny fraction of the languages supported by tesseract. Your documentation doesn't explicitly say that it does run on tesseract (so I'm not suggesting that it's misleading). Rather, mention of this should probably be made in the summary of ScribeOCR on the "3rdParty" page I referred to above.

Could you please confirm that my understanding is correct? If so, I'm sure I'll be able to find use for ScribeOCR for documents written in the more common languages that it supports.

Thanks!
Andy

**Comment by Balearica** [comment]
The recognition feature of Scribe OCR does indeed use Tesseract for recognition.  As Scribe OCR is a web application, it is unable to access your local Tesseract installation.  Instead, it uses [Tesseract.js](https://github.com/naptha/tesseract.js), which is the JavaScript/WebAssembly build of Tesseract.  

Regarding language support, Scribe OCR does currently support fewer languages than Tesseract.  The limiting factor for this is not recognition (we can recognize any language Tesseract supports), but rather that the proofreading features do not yet support scripts other than Latin.  As a result, most popular languages that use Latin script are listed as supported (e.g. English, Spanish, French, German, Portuguese, etc.), and any missing languages that use Latin script can easily be added.  However, languages that do not use Latin script will require significant updates to support, which will happen at some later point.  (While Simplified Chinese is on the language list, that is essentially experimental, and not all features work.) 

As we probably get a non-trivial amount of traffic from the Tesseract page, I added a new page to the [docs](https://docs.scribeocr.com/) that should hopefully answer some common questions from Tesseract users:
https://docs.scribeocr.com/tesseract_users.html

**Comment by ajlowy66** [comment]
Thanks for your prompt response and explanation.

As you can probably guess from the nature of my question, I’m interested in (or need) non-Latin (I think you mean non-Roman) languages, i.e. Hebrew and Greek. Clearly, ScribeOCR won’t help me in the near-to-medium term, and yes, there are alternatives to ScribeOCR that do run on users’ locally installed tesseract. However, most of them are old, buggy, or just don’t work terribly well (with the exception of gImageReader). So I was just looking at all the options for those two non-Roman alphabet languages, although I do also need to translate several German documents so I look forward to using ScribeOCR for them.

Regards,
Andrew



On 25 Jun 2024, at 2:34 AM, Balearica ***@***.***> wrote:


The recognition feature of Scribe OCR does indeed use Tesseract for recognition. As Scribe OCR is a web application, it is unable to access your local Tesseract installation. Instead, it uses Tesseract.js <https://github.com/naptha/tesseract.js>, which is the JavaScript/WebAssembly build of Tesseract.

Regarding language support, Scribe OCR does currently support fewer languages than Tesseract. The limiting factor for this is not recognition (we can recognize any language Tesseract supports), but rather that the proofreading features do not yet support scripts other than Latin. As a result, most popular languages that use Latin script are listed as supported (e.g. English, Spanish, French, German, Portuguese, etc.), and any missing languages that use Latin script can easily be added. However, languages that do not use Latin script will require significant updates to support, which will happen at some later point. (While Simplified Chinese is on the language list, that is essentially experimental, and not all features work.)

As we probably get a non-trivial amount of traffic from the Tesseract page, I added a new page to the docs <https://docs.scribeocr.com/> that should hopefully answer some common questions from Tesseract users:
https://docs.scribeocr.com/tesseract_users.html
